docs/database-structure.md

```diff
@@ -0,0 +1,336 @@
+- status: Confirmation that the event finished correctly (NULL or done for now).
+- path: The path of the event (the relative url of the event).
+- campus: The campus that contains the event.
+- code: Used to unlock the exams.
+
+### event_user
+
+This table links users and events. A user can be in many events and an events have many users.
+
+Columns:
+- id
+- createdAt
+- userId
+- eventId
+- groupId (not used anymore, TODO: remove at some point)
+
+### group
+
+Some projects are required to be made by groups. Those groups are kept in this table.
+
+Columns:
+- id
+- objectId: The object that is being "done" by the group. Usually projects or raids.
+- eventId : The groups only exist in relation to an event
+- captainId: The captain of the group.
+- createdAt.
+- updatedAt.
+- status: See [group_status](#group_status) for possible values.
+- path.
+- campus.
+
+### group_status
+
+Columns:
+- status
+Possible values (for now)
+  - `setup`
+  - `working`
+  - `audit`
+  - `finished`
+
+### group_user
+
+This table links users to groups. A user can be in many groups and a group contains many users.
+
+Columns:
+- id
+- userId
+- groupId
+- confirmed: Boolean. False by default. True after the user represented by the user id confirms the group.
+- createdAt
+- updatedAt
+
+### match
+
+It is used for the `bet` system in the bonus exercises. Specifically to keep a record of the matches between students who will review and make a bet about each others code.
+
+Columns:
+- id
+- createdAt
+- updatedAt
+- objectId: The object that requires the match.
+- userId: The user that is matched.
+- matchId: The matched that is made to another user.
+- confirmed: The same as in group_user it's a way to keep track of invitations that had not been confirmed.
+- bet: boolean; For the optional exercises that require a bet. A bet can be true if the exercises is expected to pass and false otherwise.
+- result: boolean; True for correct bet.
+- path
+- campus
+- eventId: The event the object in inserted in. (Piscine, Raid, etc).
+
+### object
+
+Objects are generic representations of elements in the structure of the curriculum (Onboarding, campus, piscine, raids, etc). They are arranged in a hierarchical structure to allow unlimited nesting (see [object_child](#object_child)). They also keep the timeline of the events relaying in durations (ex. `duration`, `eventDuration`, etc) specified by [attributes](attributes.md) (`attrs`).
+
+Columns:
+- id
+- name
+- type: See [object_type](#object_type) for the possible types.
+- status: (Currently not in used, but it should always be `online` to avoid errors TODO: remove at some point).
+- attrs: See [object attributes](attributes.md).
+- childrenAttrs: Attributes that are applied to all the children objects.
+- createdAt
+- updatedAt
+- externalRelationUrl (Maybe not needed).
+- authorId (Maybe not needed).
+- campus: The campus the object is related to.
+- referenceId: A reference to the base object.
```

Comment:
if duplicated, the id of the reference or creation object

Reply:
Thank you. That's clearer